File: backend/app/routes/teachers.py
from flask import Blueprint, jsonify

from app.models.teacher import (
    get_all_teachers,
    get_teacher_by_id,
    serialize_teacher,
)

teachers_bp = Blueprint('teachers', __name__, url_prefix='/api/teachers')


# -------------------------------------------------------------------
# GET /api/teachers
# -------------------------------------------------------------------
@teachers_bp.route('', methods=['GET'])
def list_teachers():
    teachers = get_all_teachers()
    return jsonify({
        "success": True,
        "data": [serialize_teacher(t) for t in teachers],
    }), 200


# -------------------------------------------------------------------
# GET /api/teachers/<teacher_id>
# -------------------------------------------------------------------
@teachers_bp.route('/<teacher_id>', methods=['GET'])
def single_teacher(teacher_id):
    teacher = get_teacher_by_id(teacher_id)
    if not teacher:
        return jsonify({"success": False, "message": "Teacher not found"}), 404

    return jsonify({
        "success": True,
        "data": serialize_teacher(teacher),
    }), 200

File: backend/routes/teachers.js
import express from 'express';
import { getDB, ObjectId } from '../db.js';

const router = express.Router();

function serializeTeacher(t) {
  if (!t) return null;
  return {
    id: t._id.toString(),
    name: t.name || t.full_name || '',
    title: t.title || '',
    bio: t.bio || '',
    experience_years: t.experience_years || 0,
    image: t.image || '',
    specializations: t.specializations || [],
  };
}

// GET /api/teachers
router.get('/', async (req, res) => {
  try {
    const db = getDB();
    const teachers = await db.collection('teachers').find().toArray();
    return res.json({
      success: true,
      data: teachers.map(serializeTeacher),
    });
  } catch (err) {
    return res.status(500).json({ success: false, message: 'Server error' });
  }
});

// GET /api/teachers/:teacher_id
router.get('/:teacher_id', async (req, res) => {
  try {
    const db = getDB();
    const teacher = await db.collection('teachers').findOne({ _id: new ObjectId(req.params.teacher_id) });
    if (!teacher) {
      return res.status(404).json({ success: false, message: 'Teacher not found' });
    }
    return res.json({ success: true, data: serializeTeacher(teacher) });
  } catch (err) {
    return res.status(500).json({ success: false, message: 'Server error' });
  }
});

export default router;
